webapp/src/main/java/com/box/l10n/mojito/service/blobstorage/redis/RedisScriptManager.java

```diff
@@ -0,0 +1,82 @@
+package com.box.l10n.mojito.service.blobstorage.redis;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.ResourceLoader;
+import org.springframework.stereotype.Component;
+import redis.clients.jedis.Jedis;
+
+/**
+ * Reads and loads all lua scripts defined in the {@link RedisScript} enum into Redis and stores
+ * their SHA hashes for later execution.
+ *
+ * <p>Scripts are loaded at application startup and can be executed using their enum representation.
+ * To execute a script, use {@link RedisClient#executeScript(RedisScript, List, List)}.
+ *
+ * @author mwilshire
+ */
+@Component
+@ConditionalOnProperty("l10n.redis.connection.endpoint")
+public class RedisScriptManager {
+  static Logger logger = LoggerFactory.getLogger(RedisScriptManager.class);
+
+  private final String SCRIPTS_PATH = "redis/scripts";
+  private final ResourceLoader resourceLoader;
+  private final RedisPoolManager redisPoolManager;
+
+  private final Map<RedisScript, String> scriptToSHA = new HashMap<>();
+
+  public RedisScriptManager(ResourceLoader resourceLoader, RedisPoolManager redisPoolManager) {
+    this.resourceLoader = resourceLoader;
+    this.redisPoolManager = redisPoolManager;
+    Arrays.stream(RedisScript.values()).forEach(this::loadScript);
+  }
+
+  public void loadScript(RedisScript script) {
+    try {
+      String scriptContent = loadScriptContent(script);
+
+      try (Jedis jedis = redisPoolManager.getJedis()) {
+        String scriptSha = jedis.scriptLoad(scriptContent);
+        scriptToSHA.put(script, scriptSha);
+        logger.info(
+            "Script '{}' loaded into Redis with SHA: {}", script.getScriptName(), scriptSha);
+      }
+    } catch (Exception e) {
+      logger.error("Failed to load script {}: {}", script.getScriptName(), e.getMessage(), e);
+    }
+  }
+
+  private String loadScriptContent(RedisScript script) throws IOException {
+    String resourcePath = "classpath:" + SCRIPTS_PATH + "/" + script.getScriptName();
```

Comment:
:nit: Restricting this to classpath means the script needs to be built into the JAR, could we expand this to be a configurable path to allow for more flexibility? Users could then use whatever lua script implementation they want as long as it meets the contract of returning 1 for allowed and 0 for not allowed?  Probably a bit outside the scope of this ticket.

I guess the classpath restriction is to allow 'typing' of what rate limiter to use in other areas of the code